### ISSUE TITLE
Variable referencing

### ISSUE THREAD

**Issue by nikolay** [progrium/bashstyle#24]
How about standardizing on braces?
```bash
# Yep
echo "${var}"
echo "$1"

# Nope
echo "$var"
```

**Comment by progrium** [comment]
This is a common convention that I've not yet adopted. I understand the argument for consistency and avoiding surrounding text from being interpreted as part of the name. It's just one of those things that's so different from de facto Bash I'm hesitant. I'll leave this open and if I ever use it in a project and it feels right, I'll adopt it here. 

Thanks for bringing this one up.

**Comment by stuartpb** [comment]
Yeah, I've found that this is generally overkill, since I'm practically never expanding a variable next to a character that's a valid identifier character. I prefer curly-brace expansion as a signifier of "I'm doing something crazy with this variable" like replacement or substringing.

But, then, I use backticks for substituting non-nested commands, so maybe I'm just lazy / character-miserly.

**Comment by ernstki** [comment]
It adds unnecessary visual noise, and is not required for simple variable interpolation most of the time.

It's only _really_ required when you need to stand off the variable name from a string suffix, _e.g._, `"${base}_suffix.ext"`, or `"${RED}ERROR!$RESET"`.

But…

…when you list off all the other times when braces _are_ required around a variable name—parameter expansion (`${param:+default}`) and arrays (`"${array[@]}"`)—I can definitely start to see where you're coming from, from the perspective of consistency.

**Comment by nikolay** [comment]
Conventions are about picking the best way among multiple options by the spec.

**Comment by meleu** [comment]
my two cents: I like to not waste time thinking about syntax/styles while coding. In this context, I mean I prefer to always use `${curly_braces}` for consistency **and** just to avoid ":thinking: should I use curly braces here?"

**Comment by nikolay** [comment]
@meleu Yup! Conventions are not for the lazy but for the consistent.
